### PR TITLE
Remove unused Sample Transmission c++ interface

### PR DESCRIPTION
**Pull request by robertapplin**
**Description of work.**
This PR removes the old C++ Sample Transmission interface code which is no longer in use. It also removes some other files in the `scientific_interfaces/General` folder which are unused.

**To test:**
Code review. Make sure the new python implementation of Sample Transmission Calculator interface still appears in the `Interfaces`->`General` menu

*There is no associated issue.*

*This does not require release notes* because **it is internal maintenance**

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/scientific_interfaces/General/CMakeLists.txt [hunk 1]
```diff
@@ -1,20 +1,20 @@
-set(SRC_FILES MantidEV.cpp MantidEVWorker.cpp SampleTransmission.cpp StepScan.cpp)
+set(SRC_FILES FitScriptGenerator.cpp StepScan.cpp)
 
 # Include files aren't required, but this makes them appear in Visual Studio IMPORTANT: Include files are required in
 # the MOC_FILES set. Scroll down to find it.
-set(INC_FILES DllConfig.h MantidEV.h MantidEVWorker.h SampleTransmission.h StepScan.h Updateable.h)
```
Comment (ConorMFinn) [inline qt/scientific_interfaces/General/CMakeLists.txt:5]: I notice `StepScan.h` has been removed here but not for the `SRC_FILES`. Is this intentional?

`FitScriptGenerator.h` also isn't here. Is the comment about the files not appearing in VS no longer the case?

**Review by cailafinn — CHANGES_REQUESTED** [review]
Calculator still works properly, nice to get this removed.

Just a couple of questions on the CMake

File: qt/scientific_interfaces/General/CMakeLists.txt (hunk 1 above)
Comment (robertapplin) [reply qt/scientific_interfaces/General/CMakeLists.txt:5]: They do still appear in VS, so this comment must be outdated. I have updated the comment to make it clearer that only non-Qt header files need to be specified in the `INC_FILES` variable

**Review by cailafinn — APPROVED** [review]
Looks good to me, nice to tidy up the codebase 👍

**Pull request merged**